### PR TITLE
Migrate: Resource group resource to Platform-go-sdk

### PR DESCRIPTION
**Pull request by kavya498**

### CODE REVIEW

File: ibm/config.go [hunk 1]
```diff
@@ -1568,6 +1578,16 @@ func (c *Config) ClientSession() (interface{}, error) {
...
 
+	resourceManagerOptions := &resourcemanager.ResourceManagerV2Options{
+		Authenticator: authenticator,
+		URL:           envFallBack([]string{"IBMCLOUD_RESOURCE_MANAGER_API_ENDPOINT"}, "https://resource-controller.cloud.ibm.com/v2"),
+	}
+	resourceManagerClient, err := resourcemanager.NewResourceManagerV2(resourceManagerOptions)
+	if err != nil {
+		session.resourceManagerErr = fmt.Errorf("Error occured while configuring Resource Manager service: %q", err)
+	}
+	session.resourceManagerAPI = resourceManagerClient
```
Comment (hkantare) [inline ibm/config.go:1589]: Add enable Retries function

File: ibm/resource_ibm_resource_group.go [hunk 2]
```diff
@@ -58,81 +97,128 @@ func resourceIBMResourceGroupCreate(d *schema.ResourceData, meta interface{}) er
...
-		},
+	resourceGroupCreate := rg.CreateResourceGroupOptions{
+		Name:      &name,
+		AccountID: &accountID,
 	}
 
-	resourceGroup, err := rMgtClient.ResourceGroup().Create(resourceGroupCreate)
+	resourceGroup, resp, err := rMgtClient.CreateResourceGroup(&resourceGroupCreate)
 	if err != nil {
-		return fmt.Errorf("Error creating resource group: %s", err)
```
Comment (hkantare) [inline ibm/resource_ibm_resource_group.go:73]: Print complete resp

File: ibm/config.go (hunk 1 above)
Comment (kavya498) [reply ibm/config.go:1590]: done

File: ibm/resource_ibm_resource_group.go (hunk 2 above)
Comment (kavya498) [reply ibm/resource_ibm_resource_group.go:73]: done

**Pull request merged**